libcontainer/landlock/config.go

```diff
@@ -0,0 +1,31 @@
+package landlock
+
+import (
+	"fmt"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+var accessFSs = map[string]configs.AccessFS{
+	"execute":     configs.Execute,
+	"write_file":  configs.WriteFile,
+	"read_file":   configs.ReadFile,
+	"read_dir":    configs.ReadDir,
+	"remove_dir":  configs.RemoveDir,
+	"remove_file": configs.RemoveFile,
+	"make_char":   configs.MakeChar,
+	"make_dir":    configs.MakeDir,
+	"make_reg":    configs.MakeReg,
+	"make_sock":   configs.MakeSock,
+	"make_fifo":   configs.MakeFifo,
+	"make_block":  configs.MakeBlock,
+	"make_sym":    configs.MakeSym,
+}
+
+// ConvertStringToAccessFS converts a string into a Landlock access right.
+func ConvertStringToAccessFS(in string) (configs.AccessFS, error) {
+	if access, ok := accessFSs[in]; ok {
+		return access, nil
+	}
+	return 0, fmt.Errorf("string %s is not a valid access right for landlock", in)
+}
```

Comment:
Would be nice to have this functionality in https://github.com/landlock-lsm/go-landlock/blob/main/landlock/accessfs.go and use it.

Reply:
Related to https://github.com/landlock-lsm/go-landlock/issues/14

Reply:
Feel free to create a PR for go-landlock :wink:

Reply:
As I mentioned also in https://github.com/landlock-lsm/go-landlock/issues/14, please be aware that when this string-to-enum mapping is part of go-landlock, the set of supported values might grow at some point in the future when you update the go-landlock library, and this might change the permitted values in your configuration language implicitly.

If this is the approach you want to take, and since @l0kod seems to also be on board, we could make that mapping part of the library, in the lower-case, underscore-delimited variant ("write_file")

Reply:
It may help to have an helper to check that an access right is supported by a specific ABI version, or specify the version as an argument. I'm not sure if the string conversion belongs to go-landlock or libcontainer though. Is libcontainer used by other container runtimes?

Reply:
It should be usable for other container runtimes but I'm not sure here. @kolyshkin @AkihiroSuda @cyphar Any insights? Thanks!